### PR TITLE
Grafana sidecar for loading dynamically dashboards

### PR DESCRIPTION
**Pull request by lzecca78**
Hi everyone 👋 !
i've implemented the `k8s-sidecar` that allow everyone to eventually inject in grafana whatever dashboard just creating a configmap in his namespace and labelling it with the label-key defined in the deployment. (look at README.md for more details)
This  make massively easier to add/remove/modify new/exisisting dashboard without blood and tears.
Looking forward for your review!

### CODE REVIEW

**Comment by lnovara** [comment]
[Improve Grafana manifests](https://trello.com/c/Hy7EQmOT/17-improve-grafana-manifests)

**Comment by lzecca78** [comment]
@phisco the right way is the linter one. Once that all files will be linted, you'll never see any modifications 👨‍🔧

**Comment by phisco** [comment]
That’s the case if we all agree on a linter and/or spec, otherwise your linter will just keep removing spaces and white lines and mine (I don’t have one for yaml) could just keep adding them back 🤷🏻‍♂️ I love go fo having an official linter, that can be automated on save, yaml ones are just a matter of taste

File: katalog/grafana/README.md [hunk 1]
```diff
@@ -3,63 +3,65 @@
...
 numeric time-series data with Prometheus integration.
 
-
 ## Image repository and tag
 
-* Grafana image: `grafana/grafana:6.6.2`
-* Grafana repository: https://github.com/grafana/grafana
-* Grafana documentation: https://docs.grafana.org
-
+-   Grafana image: `grafana/grafana:6.6.2`
```
Comment (angelbarrera92) [inline katalog/grafana/README.md:8]: Why the `* ` has been disappeared?

File: katalog/grafana/README.md [hunk 2]
```diff
@@ -3,63 +3,65 @@
...
+-   k8s-sidecar documentation: https://github.com/kiwigrid/k8s-sidecar/blob/master/README.md
 
 ## Requirements
 
-- Kubernetes >= `1.10.0`
-- Kustomize = `v1.0.10`
-
-
+-   Kubernetes >= `1.10.0`
+-   Kustomize = `v1.0.10`
```
Comment (angelbarrera92) [inline katalog/grafana/README.md:18]: Please, change to v3

Comment (angelbarrera92) [reply katalog/grafana/README.md:18]: And Kubernetes >1.14

**Comment by lzecca78** [comment]
@phisco i guess that is worst going on without linter because at least, linter has a standard, no linter means that anyone can do whatever it likes, that means "mess".

File: katalog/grafana/README.md (hunk 1 above)
Comment (lzecca78) [reply katalog/grafana/README.md:8]: because is a duplication of the `-` list in markdown.

File: katalog/grafana/README.md (hunk 2 above)
Comment (lzecca78) [reply katalog/grafana/README.md:18]: 👍

**Comment by ralgozino** [comment]
Now that the linting issue has been addressed and we've agreed on using police-man could you please add it to the pipeline @lzecca78 and let us know so we can move on with the review?
Thanks!

**Comment by lzecca78** [comment]
@ralgozino the modification that you are requesting will impact `all` the module monitoring that is the grafana dir `plus` everything else. This is imho out of scope with the aim of this PR. If you want, we can create an issue that will be converted in an activity, but i guess that is totally unrelated with this one.

**Comment by ralgozino** [comment]
as a minimum, I would say to apply police-man to the files that you are modifying in this PR, it is official since yesterday so we have to start enforcing it.
I don't know if it makes sense for the rest of the repo, I guess it depends on how many changes it will trigger, maybe the files are already mostly compliant with the linting rules.
Thoughts @angelbarrera92 @iknite @phisco ?

**Comment by angelbarrera92** [comment]
I included policeman in a couple of modules and it requires some time to fix (possible) issues. But we are talking about.... 30 min? So i think is worthy to include it in the PR.

If you don't want, i can do it for you. Just let me know Luca

**Comment by lzecca78** [comment]
Ok I will do it tomorrow!

**Comment by lzecca78** [comment]
@angelbarrera92 @ralgozino , linting done

**Comment by lzecca78** [comment]
@angelbarrera92 added dependencies and modified image version in `.drone.yml`

File: katalog/grafana/kustomization.yaml [hunk 3]
```diff
@@ -1,111 +1,23 @@
+---
 namespace: monitoring
 
+bases:
```
[the remark this answers is not in the document]
Comment (angelbarrera92) [reply katalog/grafana/kustomization.yaml:4]: Understood ;)

File: .drone.yml [hunk 4]
```diff
@@ -52,12 +74,12 @@ trigger:
 
 steps:
   - name: init
-    image: quay.io/sighup/e2e-testing-drone-plugin:v0.4.2
+    image: quay.io/sighup/e2e-testing-drone-plugin:v0.5.0
```
Comment (angelbarrera92) [inline .drone.yml:77]: Nice!

File: .drone.yml [hunk 5]
```diff
@@ -92,7 +114,7 @@ steps:
   - name: destroy
     image: quay.io/sighup/e2e-testing-drone-plugin:v0.4.2
```
Comment (angelbarrera92) [inline .drone.yml:115]: I said to replace v0.4.2 -> v0.5.0 in all appearance of `quay.io/sighup/e2e-testing-drone-plugin:v0.4.2`.
Can you do it?

**Pull request merged**